haddock-library/test/Documentation/Haddock/ParserSpec.hs

```diff
@@ -86,6 +86,19 @@ spec = do
       it "parses identifiers enclosed within backticks" $ do
         "`foo`" `shouldParseTo` DocIdentifier "foo"
 
+      it "parses identifiers preceded by a backtick and followed by a single quote" $ do
+        "`foo'" `shouldParseTo` DocIdentifier "foo"
+
+      -- This one is funky.
+      it "parses identifiers preceded by a single quote and followed by a backtick" $ do
+        "'foo`" `shouldParseTo` DocIdentifier "foo"
```

Comment:
This syntax isn't mentioned in http://haskell-haddock.readthedocs.io/en/latest/markup.html#hyperlinked-identifiers but it works nonetheless. Not sure if it should be legal or not.

Reply:
Maybe we should add docs for that then? Does it have to do with quoting of identifiers ending with '''?

Reply:
Huh, I'm sure I had written a response to your comment, but I must have forgotten to submit it. :/

> Does it have to do with quoting of identifiers ending with '''?

I don't think so. The identifier parser simply accepts any combination of delimiters:

https://github.com/haskell/haddock/blob/a8840a37976209c5a3c4cf929eff502cdca80ecf/haddock-library/src/Documentation/Haddock/Parser.hs#L849-L856

> Maybe we should add docs for that then?

I'm not convinced that we should let the existing parser guide what we define as legal syntax.